tests/components/airthings/test_sensor.py

```diff
@@ -0,0 +1,111 @@
+"""Test the Airthings sensors."""
+
+from airthings import API_URL
+
+from homeassistant.components.airthings import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from . import TEST_DATA
+
+from tests.common import MockConfigEntry, load_fixture
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def init_integration_with_devices(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> MockConfigEntry:
+    """Set up the Airly integration in Home Assistant."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=TEST_DATA,
+    )
+
+    aioclient_mock.post(
+        "https://accounts-api.airthings.com/v1/token",
+        text=load_fixture("token.json", DOMAIN),
+    )
+    aioclient_mock.get(
+        API_URL + "locations", text=load_fixture("locations.json", DOMAIN)
+    )
+    aioclient_mock.get(API_URL + "devices", text=load_fixture("devices.json", DOMAIN))
+    aioclient_mock.get(
+        API_URL + "/locations/00000000-0000-0000-0000-000000001/latest-samples",
+        text=load_fixture("location-samples.json", DOMAIN),
+    )
```

Comment:
Ideally we mock out the client. This way we are not dependent on the device specifics

Reply:
Thank you for the feedback, I updated the PR. 🙏